packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -204,10 +319,153 @@ export async function provideTspconfigCompletionItems(
               return item;
             });
             result.push(...enums);
+          } else if (
+            cur.type === "string" &&
+            sourceType === "QUOTE_DOUBLE" &&
+            /{(env\.)?}|{[^{}]*}/g.test(source)
+          ) {
+            // Variable interpolation
+            // environment-variables
+            if (/{env\.}(?!\S)|{env\.}/g.test(source)) {
+              for (const env of target.envs) {
+                if (!nodePath.includes(env)) {
+                  result.push({
+                    label: env,
+                    kind: CompletionItemKind.Value,
+                    documentation: cur.description,
+                  });
+                }
+              }
+            } else {
+              // built-in variables
+              result.push(
+                ...["cwd", "project-root"].map((value) => {
+                  const item: CompletionItem = {
+                    label: value,
+                    kind: CompletionItemKind.Value,
+                    documentation: cur.description,
+                  };
+                  return item;
+                }),
+              );
+              // parameters
+              for (const param of target.parameters) {
+                if (!nodePath.includes(param)) {
+                  result.push({
+                    label: param,
+                    kind: CompletionItemKind.Value,
+                    documentation: cur.description,
+                  });
+                }
+              }
+            }
+          } else if (cur.type === "string") {
+            // extends
+            if (relativeFiles.length > 0) {
+              for (const file of relativeFiles) {
+                if (target.siblings.includes(file)) {
+                  continue;
+                }
+
+                const item: CompletionItem = {
+                  label: file,
+                  kind: CompletionItemKind.Field,
+                };
+                result.push(item);
+              }
+            }
           }
         }
       });
 
     return distinctArray(result, (t) => t.label);
   }
 }
+
+/**
+ * Get the new text value
+ * @param sourceType input quote type(single, double, none)
+ * @param formatText format text value
+ * @returns
+ */
+function getNewTextValue(sourceType: string, formatText: string): string {
```

Comment:
please define a specific SourceType type

Reply:
In order to solve this issue: emitter option auto complete while inside "" will add extra ""`, the getNewTextValue method was originally intended for this purpose